### PR TITLE
privoxy: Add Privoxy tools, fix https

### PR DESCRIPTION
**Pull request by essandess**
* Addresses: https://lists.privoxy.org/pipermail/privoxy-users/2021-November/000597.html

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [x] security fix

###### Tested on

macOS 11.6.1 20G224 x86_64
Xcode 13.1 13A1030d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: www/privoxy/Portfile
Comment (pmetzger) [inline www/privoxy/Portfile:85]: What's the purpose of this big block of commented out lines?

**Review by pmetzger — COMMENTED** [review]
Someone else should review this, I only have one (minor) comment myself.

File: www/privoxy/Portfile
Comment (essandess) [reply www/privoxy/Portfile:85]: The next time I or someone needs to generate a patch from the upstream’s configuration files, those are the commands that accomplish it. Easier than doing all that from scratch.

Comment (pmetzger) [reply www/privoxy/Portfile:85]: Then the comment should explain that this is what is being described. You should have a description saying "this is how you generate a patch" or whatever would otherwise give someone in the future enough information to follow in your footsteps. As it stands, the commands appear in comments without explanation.

Comment (essandess) [reply www/privoxy/Portfile:85]: Thanks, done.

**Review by mascguy — APPROVED** [review]

**Comment by essandess** [comment]
@mascguy @pmetzger Please note that I just [force-pushed](https://github.com/macports/macports-ports/compare/7a9d14c782a860958a2afbe94f87dea639a4a5b6..36f29cf56a889a9e9fe9eef00d2599022564d435) commented-out options to use OpenSSL3's preferred command to generate RSA private keys.

**Comment by mascguy** [comment]
Steve, let us know when your changes are done.

In the interim, I'll keep cancelling old CI jobs for this PR, when they're superseded by later commits. Going forward though, to avoid clogging the CI queue, you can add `[ci skip]` to your commit message to avoid unnecessary CI runs. (At least for those times when you don't need immediate CI feedback.)

Then, when your changes are complete, make one final push without that. Documented here:

https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/

File: www/privoxy/Portfile
Comment (pmetzger) [inline www/privoxy/Portfile:339]: I'm not clear on what this is documenting? Is it purely for the maintainer? If it is for the end user, it probably is better to explain it in a message on package install.

File: www/privoxy/Portfile
Comment (pmetzger) [inline www/privoxy/Portfile:432]: Same comment as above.

**Review by pmetzger — COMMENTED** [review]
More minor comments.

File: www/privoxy/Portfile
Comment (essandess) [reply www/privoxy/Portfile:339]: It's just an obvious choice between `EC` and `RSA` certificates, and should be self-explanatory to anyone managing their PKI. The default here is EC, but the RSA option is there if anyone wants it. I think this falls into the category "if you have to ask…".

File: www/privoxy/Portfile
Comment (essandess) [reply www/privoxy/Portfile:432]: Same as above.

**Comment by essandess** [comment]
I'm done with edits for this PR, subject to @pmetzger's approval. Sorry for clogging the pipeline—I thought that stuff was automatically cancelled.

**Comment by mascguy** [comment]
> I'm done with edits for this PR, subject to @pmetzger's approval. Sorry for clogging the pipeline—I thought that stuff was automatically cancelled.

No worries, most folks aren't aware of the behavior. And feel free to spread the word!

File: www/privoxy/Portfile
Comment (pmetzger) [reply www/privoxy/Portfile:339]: Even sophisticated users don't read portfiles. I'd do another pull with some instructions in the install messages. Not strictly necessary of course.

**Pull request merged**
